### PR TITLE
Pj direct 2 es

### PR DESCRIPTION
**Pull request by peterj99a**
Add setting to by pass AWS when indexing

### CODE REVIEW

File: stack/core/src/main/java/org/apache/usergrid/corepersistence/service/CollectionServiceImpl.java [hunk 1]
```diff
@@ -58,8 +59,12 @@ public CollectionServiceImpl( final PipelineBuilderFactory pipelineBuilderFactor
         final Optional<String> query = search.getQuery();
 
         final IdBuilder pipelineBuilder =
-            pipelineBuilderFactory.create( applicationScope ).withCursor( search.getCursor() )
-                                  .withLimit( search.getLimit() ).fromId( search.getCollectionOwnerId() );
+            pipelineBuilderFactory.create( applicationScope )
+                .withCursor( search.getCursor() )
+                .withLimit( search.getLimit() )
+                .keepStaleEntries(search.getKeepStaleEntries())
+                .query(search.getQuery())
```
Comment (mdunker) [inline stack/core/src/main/java/org/apache/usergrid/corepersistence/service/CollectionServiceImpl.java]: .query(query)

File: stack/corepersistence/queue/src/main/java/org/apache/usergrid/persistence/queue/settigs/QueueIndexingStrategy.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
+/*
```
Comment (mdunker) [inline stack/corepersistence/queue/src/main/java/org/apache/usergrid/persistence/queue/settigs/QueueIndexingStrategy.java]: .../settigs/QueueIndexingStrategy -- this is a duplicate file

File: stack/corepersistence/queue/src/main/java/org/apache/usergrid/persistence/queue/settigs/IndexConsistency.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
+/*
```
Comment (mdunker) [inline stack/corepersistence/queue/src/main/java/org/apache/usergrid/persistence/queue/settigs/IndexConsistency.java]: .../settigs/IndexConsistency -- this is a duplicate file

File: stack/core/src/main/java/org/apache/usergrid/corepersistence/asyncevents/direct/BufferedQueueImpl.java [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+import java.util.concurrent.*;
+import java.util.function.Consumer;
+
+/**
+ * Bufferes events and dispatched then in batches.
+ * Ensures that the callback will be called at a min interval.
+ */
+public class BufferedQueueImpl<T> implements BufferedQueue<T> {
+
+    private String fileName = "my_file_name.txt";
```
[the remark this answers is not in the document]
Comment (peterj99a) [reply stack/core/src/main/java/org/apache/usergrid/corepersistence/asyncevents/direct/BufferedQueueImpl.java]: done

File: stack/core/src/main/java/org/apache/usergrid/corepersistence/asyncevents/AsyncEventServiceImpl.java [hunk 5]
```diff
@@ -704,7 +707,24 @@ public void queueIndexOperationMessage(final IndexOperationMessage indexOperatio
         offerTopic( elasticsearchIndexEvent, queueType );
     }
 
-    private void handleIndexOperation(final ElasticsearchIndexEvent elasticsearchIndexEvent)
+    protected ElasticsearchIndexEvent getIndexOperationMessage(final IndexOperationMessage indexOperationMessage) {
```
[the remark this answers is not in the document]
Comment (peterj99a) [reply stack/core/src/main/java/org/apache/usergrid/corepersistence/asyncevents/AsyncEventServiceImpl.java]: done

File: stack/core/src/main/java/org/apache/usergrid/corepersistence/asyncevents/direct/BufferedQueueImpl.java [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+    private Consumer<List<T>> consumer;
+
+    ScheduledThreadPoolExecutor threadPool = new ScheduledThreadPoolExecutor(1);
```
[the remark this answers is not in the document]
Comment (peterj99a) [reply stack/core/src/main/java/org/apache/usergrid/corepersistence/asyncevents/direct/BufferedQueueImpl.java]: removed

File: stack/core/src/main/java/org/apache/usergrid/corepersistence/asyncevents/direct/DirectFirstEventServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+        indexDebugMode = Boolean.valueOf(queueFig.getQueueDebugMode());
+
+    }
+
+    protected void dispatchToES(final List<Serializable> bodies) {
+
+        List<LegacyQueueMessage> messages = new ArrayList<>();
+        for (Serializable body : bodies) {
+            String uuid = UUID.randomUUID().toString();
+            LegacyQueueMessage message = new LegacyQueueMessage(uuid, "handle_" + uuid, body, "put type here");
```
[the remark this answers is not in the document]
Comment (peterj99a) [reply stack/core/src/main/java/org/apache/usergrid/corepersistence/asyncevents/direct/DirectFirstEventServiceImpl.java]: Yeah the "put type here" was in the original implementation.  I just copied it over to keep it the same

File: stack/core/src/main/java/org/apache/usergrid/corepersistence/index/settings/QueueIndexingStrategy.java [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+
+/**
+ * This class describes the paths an index request can take
+ * between tomcat and ES.
+ *
+ * Created by peterajohnson on 10/30/17.
+ */
+public enum QueueIndexingStrategy {
+
+    NOINDEX("noindex"),        // Do not Index the entity (DEBUG only use for testing)
```
[the remark this answers is not in the document]
Comment (peterj99a) [reply stack/core/src/main/java/org/apache/usergrid/corepersistence/index/settings/QueueIndexingStrategy.java]: done in the correct version of the class

File: stack/core/src/main/java/org/apache/usergrid/corepersistence/pipeline/read/search/CandidateEntityFilter.java [hunk 9]
```diff
@@ -217,6 +221,11 @@ private boolean nestedFieldCheck( String[] parts, Map<String, Field> fieldMap) {
     }
 
 
+
+    static {
```
[the remark this answers is not in the document]
Comment (peterj99a) [reply stack/core/src/main/java/org/apache/usergrid/corepersistence/pipeline/read/search/CandidateEntityFilter.java]: yeah gone

File: stack/corepersistence/queue/src/main/java/org/apache/usergrid/persistence/queue/settigs/IndexConsistency.java (hunk 3 above)
Comment (peterj99a) [reply stack/corepersistence/queue/src/main/java/org/apache/usergrid/persistence/queue/settigs/IndexConsistency.java]: removed

File: stack/corepersistence/queue/src/main/java/org/apache/usergrid/persistence/queue/settigs/QueueIndexingStrategy.java (hunk 2 above)
Comment (peterj99a) [reply stack/corepersistence/queue/src/main/java/org/apache/usergrid/persistence/queue/settigs/QueueIndexingStrategy.java]: removed

File: stack/core/src/main/java/org/apache/usergrid/corepersistence/asyncevents/direct/DirectFirstEventServiceImpl.java (hunk 7 above)
[the remark this answers is not in the document]
Comment (mdunker) [reply stack/core/src/main/java/org/apache/usergrid/corepersistence/asyncevents/direct/DirectFirstEventServiceImpl.java]: ok, missed that, sorry

**Pull request merged**